docs/guides/adapters.md

```diff
@@ -0,0 +1,115 @@
+
+checkout [a quick side note on how the DraftJS `block`, `entityRanges` and `entityMap` works, in the context of the TranscriptEditor component](./draftjs-blocks-entityrange-entitmap.md). Or feel free to skip this and come back later to it, if you are not interested in the underlying implementation.
+
+## Steps
+
+In your branch 
+
+- [ ] Create a folder with the name of the STT service - eg `speechmatics`
+- [ ] add a `adapters/${sttServiceName}/sample` folder
+- [ ] add a sample json file from the STT service in this last folder - this will be useful for testing.
+<!-- TODO: we should check these json are excluded from the bundle -->
+- [ ] add option in [adapters/index.js](adapters/index.js)
+
+In the adapters [adapters/index.js](adapters/index.js) in the  `sttJsonAdapter` function switch statement add a new `case` with the new STT service type eg `speechmatics`
+
+<!-- TODO: modify import path if module is moved/refactored -->
+```js
+import speechmaticsToDraft from './speechmatics/index';
+
+...
+
+case 'speechmatics':
+      blocks = speechmaticsToDraft(transcriptData);
+      return { blocks, entityMap: createEntityMap(blocks) };
+```
+
+- [ ] add an adapter function.
+
+as shown in the example you'd also need to add a function with the stt provider name +`ToDraft` eg `speechmaticsToDraft`that takes in the transcript data.
+
+- [ ] create a function to convert the STT data structure into draftJs blocks and entityRanges.
+
+You can see examples from `bbc-kaldi` and `autoEdit2` adapters.
+
+In pseudocode it's reccomended to follow this approach:
+
+1. Expose one function call that takes in the stt json data
+2. have a helper function `groupWordsInParagraphs` that as the name suggests groups words list from the STT provider transcript based on punctuation. and returns an array of words objects.
+
+The underlying details for this will vary depending on how the STT json of the provider present the data, and how the attributes are named etc..
+
+3. iterate over the paragraphs to create draftJS content blocks (see `bbc-kaldi` and `autoEdit2` example).
+
+```js
+wordsByParagraphs.forEach((paragraph, i) => {
+    const draftJsContentBlockParagraph = {
+      text: paragraph.text.join(' '),
+      type: 'paragraph',
+      data: {
+        speaker: `TBC ${ i }`,
+        words: paragraph.words, //
+        start: paragraph.words[0].start//
+      },
+      // the entities as ranges are each word in the space-joined text,
+      // so it needs to be compute for each the offset from the beginning of the paragraph and the length
+      entityRanges: generateEntitiesRanges(paragraph.words, 'text'), // wordAttributeName
+    };
+    // console.log(JSON.stringify(draftJsContentBlockParagraph,null,2))
+    results.push(draftJsContentBlockParagraph);
+  });
+
+```
+
+4. and use the helper function `generateEntitiesRanges` to  add the `entityRanges` to each block. - see above
```

Comment:
Tiny tiny thing but there's general inconsistencies with capitalisation etc. for your sentences, headings etc.